solutions/cses-1662.mdx

```diff
@@ -128,4 +126,31 @@ public class subarrayDivisibility {
 ```
 </JavaSection>
 	
+
+<PySection>
+
+```py 
+n = int(input())
+Z = map(int, input().split())
+
+M = [0] * n
+psums = 0
+M[psums] = 1
+for a in Z:
+	psums+= a
+	# Remember to account for negative sums
+	M[(psums % n + n) % n]+= 1
```

Comment:
The `+n` does not do anything since you `%n` right after.

Note `%` in python is modulo, while `%` in C++ and Java are remainder.

Reply:
The % operator is the modulo, which returns the remainder rather than the quotient after division.

Reply:
The % operator is the modulo, which returns the remainder rather than the quotient after division.

Reply:
I made the comment because in Python, `(psums % n + n) % n` does the exact same thing as `psums % n`.

In Python `psums % n` will always be nonnegative (assuming `n` is positive), so the extra steps you do here are not necessary.